subsys/net/l2/ethernet/ethernet.c

```diff
@@ -537,6 +549,8 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 	}
 
 error:
+	ethernet_remove_header_buf(pkt);
```

Comment:
I don't see this being defined in subsys/net of the master.

And it should belong to even 3rd commit.